action_transaction.go

```diff
@@ -387,6 +389,39 @@ func (c *Client) RevertTransaction(ctx context.Context, id string) error {
 	return err
 }
 
+// UpdateTransaction will update the broadcast callback transaction info, like: block height, block hash, status, bump.
+func (c *Client) UpdateTransaction(ctx context.Context, callbackResp *broadcast.SubmittedTx) error {
+	bump, err := bc.NewBUMPFromStr(callbackResp.MerklePath)
+	if err != nil {
+		c.options.logger.Err(err).Msgf("failed to parse merkle path from broadcast callback - tx: %v", callbackResp)
+		return err
+	}
+
+	txInfo := &chainstate.TransactionInfo{
+		BlockHash:   callbackResp.BlockHash,
+		BlockHeight: callbackResp.BlockHeight,
+		ID:          callbackResp.TxID,
+		TxStatus:    callbackResp.TxStatus,
+		BUMP:        bump,
+		// it's not possible to get confirmations from broadcast client; zero would be treated as "not confirmed" that's why -1
+		Confirmations: -1,
+	}
+
+	tx, err := c.GetTransaction(ctx, "", txInfo.ID)
+	if err != nil {
+		c.options.logger.Err(err).Msgf("failed to get transaction by id while processing callback: %v", txInfo.ID)
+		return err
+	}
+
+	tx.setChainInfo(txInfo)
+	if err = tx.Save(ctx); err != nil {
```

Comment:
You should update SyncTransaction model otherwise, we will try to check the state of updated transaction during next scheduled trigger.

Actually I would suggest to make this function a common code used in callbacks and in scheduled synchronisation (so actually move there and unify code that is in a sync transaction after querying chain).

Reply:
Thanks, good point. I tried to handle that.